src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Runtime.Augments
+{
+    public sealed partial class RuntimeThread
+    {
+        [ThreadStatic]
+        private static RuntimeThread t_currentThread;
+
+        /// <summary>
+        /// Used by <see cref="WaitHandle"/>'s multi-wait functions
+        /// </summary>
+        private WaitHandleArray<SafeWaitHandle> _waitedSafeWaitHandles;
+
+        private ThreadState _threadState;
+
+        private RuntimeThread()
+        {
+            _waitedSafeWaitHandles = new WaitHandleArray<SafeWaitHandle>(elementInitializer: null);
+            _threadState = ThreadState.Unstarted;
+
+#if PLATFORM_UNIX
+            _waitInfo = new WaitSubsystem.ThreadWaitInfo(this);
+#else
+            _waitedHandles = new WaitHandleArray<IntPtr>(elementInitializer: null);
+#endif
```

Comment:
Since we already have this as a partial class with Windows and Unix files, could we just have a FinishInitialize method or some such thing in those partial types which does this, rather than needing to ifdef on PLATFORM_UNIX as well?